solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -389,7 +389,7 @@ private void doWebMode() {
       numPagesPosted = postWebPages(args, 0, out);
       info(numPagesPosted + " web pages indexed.");
 
-    } catch (MalformedURLException e) {
+    } catch (URISyntaxException e) {
```

Comment:
URISyntaxException is only thrown by `new URI()`, the create function does not. Not sure how we should handle this.

See documentation of `URI#create(String)`: "This convenience factory method works as if by invoking the [URI(String)](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/net/URI.html#URI-java.lang.String-) constructor; any [URISyntaxException](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/net/URISyntaxException.html) thrown by the constructor is caught and wrapped in a new [IllegalArgumentException](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/lang/IllegalArgumentException.html) object, which is then thrown."

Reply:
I would make this consistent to other code and prefer a clear exception.

Reply:
I started using `URI.create` for the same reason that you just explained as it throws unchecked exception which allows us to avoid explicitly handling the `URISyntaxException`.

Should we switch to `new URI` wherever we can without changing too much? I am open to that.

Reply:
I think both is fine. I just noticed that the "catch" here looks obsolete (I wonder why compiler did not complain).

Reply:
`doWebMode()` is calling `appendUrlPath` that, as per new changes, is throwing `URISyntaxException`.

Reply:
OK, as we use new URI here, it might be better to do this also in the main code path. This would bring consistent exceptions.

Reply:
I have replaced the `URI.create` with `new URI` in `PostTool.java`.  Also replaced in few test classes as well.

Reply:
Looks great!

Reply:
> I have replaced the `URI.create` with `new URI` in `PostTool.java`. Also replaced in few test classes as well.

Actually you did this everywhere, which I am not sure if its a good idea. In PostTool i recommend to change this to get a consistent error message in both cases: for original URI and for the appendUrlPath() method.

I am not a big fan of checked Exceptions, so I would have used new URI only to make it consistent with the code around it.

Reply:
I've reverted the changes and updated it only for Post.

Reply:
```
git diff f2841ac..0007bb082 --name-only
solr/CHANGES.txt
solr/core/src/java/org/apache/solr/cli/PostTool.java
```